glyph/src/ttfp.rs

```diff
@@ -16,6 +16,46 @@ impl From<GlyphId> for owned_ttf_parser::GlyphId {
     }
 }
 
+/// A pre-rendered image of a glyph, usually used for emojis or other glyphs
+/// that can't be represented only using an outline.
+#[derive(Debug,Clone)]
+pub struct GlyphImage<'a> {
+    /// offset of the image from the normal origin (top at the baseline plus
+    /// ascent), measured in pixels at the image's current scale
+    pub origin : Point,
+    /// current scale of the image in pixels per em
+    pub scale : f32,
+    /// raw image data (not a bitmap)
+    pub data : &'a [u8],
+    /// format of the raw data
+    pub format : GlyphImageFormat,
+}
+
+impl<'a> From<owned_ttf_parser::RasterGlyphImage<'a>> for GlyphImage<'a> {
+    fn from(img : owned_ttf_parser::RasterGlyphImage<'a>) -> Self {
+        GlyphImage {
+            origin : point(img.x.into(), img.y.into()),
+            scale : img.pixels_per_em.into(),
+            data : img.data,
+            format : match img.format {
+                owned_ttf_parser::RasterImageFormat::PNG => GlyphImageFormat::PNG,
+            },
+        }
+    }
+}
+
+/// Valid formats for a [GlyphImage]
+///
+/// Note: currently only PNG images are actually returned by glyph_raster_image.
+#[non_exhaustive]
+#[derive(Debug,Clone)]
+pub enum GlyphImageFormat {
+    PNG,
+    SVG,
+    JPEG,
+    TIFF,
+}
```

Comment:
I think just having png support in a non-exhaustive enum is better than predicting the future, we can add jpeg etc when/if they become available. It also makes more sense that `GlyphImage` _itself_ is an enum, perhaps something like
```rust
#[non_exhaustive]
#[derive(Debug, Clone)]
pub enum GlyphImage<'a> {
    Png {
        origin: Point,
        scale: f32,
        data: &'a [u8],
    },
}
```

If we want to include svg here we should call the method just `Font::glyph_image` maybe, or something less rastery?

Reply:
Removing the other items from the enum makes sense; I can see that causing confusion when we don't actually support them.

I considered making the struct an enum, but I didn't like it from a user standpoint because it requires matching the enum in order to pull out non-format-specific information.  Also, if you have an image parser that auto-detects file formats, it is reasonable to just pass it the data without inspecting the format enum at all.

As far as the function name, from looking at ttf-parser, I think it is possible for a glyph to have both an SVG image and a raster image, so I think adding SVG support may be best done by adding another function to font that also returns a GlyphImage struct (and which doesn't need a scale parameter).

Reply:
Ok that makes sense. Let's just stick close to _ttf-parser_ for now then and improve later once there's more use.